### ISSUE TITLE
Local emulator needs to support SQL Authentication

### ISSUE THREAD

**Issue by atrauzzi** [Azure/azure-storage-net#607]
I'm currently running under a setup where my SQL server cannot be authenticated via Windows Authentication. This is preventing me from using the local emulator for table storage because it assumes the only way to connect to an SQL server is via named pipes.

Not going to lie, I really wish the local emulator made fewer assumptions about my local environment and was more flexible.

**Comment by mirobers** [comment]
Not sure if this helps in your case, but you can initialize the emulator to a custom SQL server. See the help for the `init` command:
`AzureStorageEmulator.exe help init`

The emulator's primary scenario is development and testing using LocalDB. Support for SQL scenarios beyond what is already supported is not on the roadmap. I suggest that you consider arafato/Azurite as an alternative if the current SQL options do not work for you.

**Comment by atrauzzi** [comment]
@mirobers - It's actually less than unhelpful, especially given that you closed this ticket without giving me an opportunity to follow up.
I'm afraid you might not have read/understood my question fully.  I did not ask you if I could use the Azure Storage Emulator for production use cases.

I asked if the emulator could be configured to connect to SQL servers using SQL server authetication, not integrated Windows/named pipes.

I've already tried reading the built-in help documentation and have already determined after trying to put in arbitrary values that it doesn't support SQL server authentication.  This isn't a support request, it's a feature request.

Just for posterity, I'm including the output of the command here because I think even as things stand, it isn't nearly specific enough:

```
Windows Azure Storage Emulator 5.3.0.0 command line tool
AzureStorageEmulator.exe init [-server serverName] [-sqlinstance instanceName] [-forcecreate|-skipcreate] [-reserveports|-unreserveports]

Performs one-time initialization to set up the emulator.
    -server serverName : Specifies the server hosting the SQL instance.
    -sqlinstance instanceName : Specifies the name of the SQL instance to be used.
    -forcecreate : Forces creation of the SQL database, even if it already exists.
    -skipcreate : Skips creation of the SQL database. This takes precedent over -forcecreate.
    -reserveports : Attempts to reserve the HTTP ports associated with the services.
    -unreserveports : Attempts to remove reservations for the HTTP ports associated with the services. This takes precedent over -reserve
ports.
    -inprocess : Performs initialization in the current process instead of spawning a new process. This requires the current process to h
ave been launched with elevated permissions.
```

As to the 3rd party implementation -- why would I use a storage option from someone other than the vendor of the technology I'm using?  That seems counter intuitive as I am using your product, not theirs.

Would you like to reopen this ticket or should I report another issue with all of this additional detail noted?  I'm pretty sure my original description was sufficient...

**Comment by mirobers** [comment]
I am aware that this is a feature request, and I'm sorry that my response seems to not have been clear enough. Let me rephrase. We don't intend to invest in this specific functionality but there is an open source project that you can use as an alternative if the current emulator/SQL options do not suffice.

I'm honestly not sure how to respond to the assertion that open source community-driven projects are counterintuitive. Let me just say that we are working closely with the maintainer of the Azurite repo and we even mention it in our official documentation.

**Comment by atrauzzi** [comment]
What's the motivation for not taking on improvements in the official emulator?
